node/src/components/consensus/highway_core/highway.rs

```diff
@@ -219,7 +219,7 @@ impl<C: Context> Highway<C> {
             self.instance_id,
         );
         self.active_validator = Some(av);
-        effects
+        self.add_new_own_vertices(effects, current_time)
```

Comment:
That's the root of the problem? We were not adding our own ping to the state?

Reply:
Yes, exactly.

And the other potential problem was that we weren't counting pings from someone else as doppelgänger pings if they weren't _much_ younger (more than one maximum round length) than the latest one we had in the state.

Reply:
Thanks 👍